conda/core/link.py

```diff
@@ -551,7 +587,35 @@ def _prepare(
                 "register", None, register_actions + history_actions, target_prefix
             )
         ]
-        return PrefixActionGroup(
+
+        # Instantiate any pre or post transactions defined by the user.
+        pre_transaction_actions, post_transaction_actions = [], []
+        for hook in context.plugin_manager.get_hook_results("pre_transaction_actions"):
+            pre_transaction_actions.append(
+                hook.action(
+                    transaction_context,
+                    target_prefix,
+                    unlink_precs,
+                    link_precs,
+                    remove_specs,
+                    update_specs,
+                    neutered_specs,
+                )
+            )
+        for hook in context.plugin_manager.get_hook_results("post_transaction_actions"):
+            post_transaction_actions.append(
+                hook.action(
+                    transaction_context,
+                    target_prefix,
+                    unlink_precs,
+                    link_precs,
+                    remove_specs,
+                    update_specs,
+                    neutered_specs,
+                )
+            )
```

Comment:
The pattern established by other plugin hooks is to define a helper method in the plugin manager to do this so we don't need to have to call `get_hook_restults` with the magic string outside of `conda.plugins.manager`:

```suggestion
        pre_transaction_actions = context.plugin_manager.get_pre_transaction_actions(
            transaction_context,
            target_prefix,
            unlink_precs,
            link_precs,
            remove_specs,
            update_specs,
            neutered_specs,
        )
        post_transaction_actions = context.plugin_manager.get_post_transaction_actions(
            transaction_context,
            target_prefix,
            unlink_precs,
            link_precs,
            remove_specs,
            update_specs,
            neutered_specs,
        )
```